engine/time/src/main/java/io/deephaven/time/TimeProviderNanoBase.java

```diff
@@ -0,0 +1,14 @@
+package io.deephaven.time;
+
+import io.deephaven.base.clock.ClockNanoBase;
+
+/**
+ * A base implementation of TimeProvider, with all methods being sourced from {@link #currentTimeNanos()}.
+ */
+public abstract class TimeProviderNanoBase extends ClockNanoBase implements TimeProvider {
+
+    @Override
+    public final DateTime currentTime() {
+        return new DateTime(currentTimeNanos());
```

Comment:
I'm strongly anti-default methods - I think it muddies the water, and potentially provides places where calls are virtualized where they otherwise wouldn't need to be.

For example, `io.deephaven.engine.table.impl.replay.DataDrivenReplayer` is explicitly based off of `DateTime`; if we accidentally forget to override `currentTime()`, then we'd still get correct, but inefficient, behavior.

Reply:
Also, it's completely acceptable to create a `TimeProvider` wholly from `System.currentTimeMillis()` if you wanted.